client/src/features/calendar/Calendar.js

```diff
@@ -39,17 +39,30 @@ const Calendar = ({ date }) => {
     const fetch = async () => {
       // Database data from server
       const response = await DataService.getAll();
+      // if you want to test the loading behavior
+      // await new Promise(resolve => setTimeout(resolve, 20_000));
       setEvents(response.data);
     };
 
-    fetch().then(setLoading(false)).catch(setLoading(false));
+    fetch()
```

Comment:
We want to pass a callback function into the `.then()` here so that things can continue to move along and the `.then` can execute asynchronously. As it was previously written, the `setLoading(false)` would fire immediately, instead of waiting for the `fetch` to complete. This resulted in our loading effect not showing.